jplag.frontend-utils/src/main/java/de/jplag/TokenPrinter.java

```diff
@@ -104,10 +106,15 @@ public static String printTokens(TokenList tokens, Collection<File> files, File
             }
 
             // New code line:
-            if (token.getLine() > lineIndex) {
-                lineIndex = token.getLine();
+            boolean previousLineEmpty = false;
+            while (token.getLine() > lineIndex) {
```

Comment:
I get that you tried to modify as little as possible with this PR and thus opted for this while loop. Before, the logic was to get the line for a token. However, now you are implicitly iterating over the lines and get the tokens for it, i.e. the logic changed to get all tokens for the current line. Thus, the code should be adapted to iterate over the lines which would also remove the need for a nested loop which is always difficult to read.

To keep the scope of this PR small, we can also make an issue and refactor the token printer in a subsequent PR. Probably @tsaglam should have the decided opinion on that 😄 .

Reply:
Both fixing it in here or making an issue is fine for me. To be fair, the token printer never was that readable to begin with. However, that's not a reason to leave it as is!